subproc_pool: Fix quiesce waitcounter

c00w opened the pull request:
Summary:
I was inspecting running jobs, and the quiesce waitcounter wasn't showing up.
Turns out this was a bad copy paste.

Test Plan: Primarily inspection

Reviewed By: masnesral

Differential Revision: D86457409




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

aorenste commented on torch/_inductor/compile_worker/subproc_pool.py:
```diff
@@ -319,11 +319,11 @@ def _read_thread(self) -> None:
+        if self.quiesce_waitcounter is None:
+            self.quiesce_waitcounter = _WaitCounter(
+                "pytorch.wait_counter.subproc_pool.quiesced"
+            ).guard()
```
(I guess the idea is that under normal circumstances it's only called once?)

c00w replied:
If you trace the logic, this should get set for each quiesce, but it goes through a cycle, we create it when we quiesce, then we close it once we get a job.

c00w committed:
subproc_pool: Fix quiesce waitcounter (#167350)

Summary:

I was inspecting running jobs, and the quiesce waitcounter wasn't showing up.
Turns out this was a bad copy paste.

Test Plan: Primarily inspection

Reviewed By: masnesral

Differential Revision: D86457409

c00w commented on torch/_inductor/compile_worker/subproc_pool.py:
```diff
@@ -319,11 +319,11 @@ def _read_thread(self) -> None:
-            "pytorch.wait_counter.subproc_pool.running"
-        ).guard()
-        self.quiesce_waitcounter.__enter__()
+        if self.quiesce_waitcounter is None:
```
This was a bad internal export with stacked diffs, fixed.
